plugins/mongodb/mongodbembed.cpp

```diff
@@ -0,0 +1,2302 @@
+     * @param name Name of the parameter.
+     * @param val VString value.
+     */
+    void MongoDBEmbedFunctionContext::bindVStringParam(const char *name, const char *val)
+    {
+        checkNextParam(name);
+
+        query->build()->append(kvp(std::string(name), bsoncxx::types::b_utf8{std::string(val)}));
+    }
+
+    /**
+     * @brief Binds an ECL UTF8 param to a bsoncxx::types::b_utf8.
+     * 
+     * @param name Name of the parameter.
+     * @param chars Number of chars in string.
+     * @param val UTF8 value.
+     */
+    void MongoDBEmbedFunctionContext::bindUTF8Param(const char *name, size32_t chars, const char *val)
+    {
+        checkNextParam(name);
+
+        query->build()->append(kvp(std::string(name), bsoncxx::types::b_utf8{std::string(val)}));
+    }
+
+    /**
+     * @brief Binds an ECL Unicode param to a bsoncxx::types::b_utf8.
+     * 
+     * @param name Name of the parameter.
+     * @param chars Number of chars in string.
+     * @param val Unicode value.
+     */
+    void MongoDBEmbedFunctionContext::bindUnicodeParam(const char *name, size32_t chars, const UChar *val) 
+    {
+        checkNextParam(name);
+        size32_t utf8chars;
+        char *utf8;
+        rtlUnicodeToUtf8X(utf8chars, utf8, chars, val);
+
+        query->build()->append(kvp(std::string(name), bsoncxx::types::b_utf8{std::string(utf8, utf8chars)}));
+    }
+
+    /**
+     * @brief Configures a mongocxx::instance and allows for multiple threads to use it for making connections.
+     * 
+     * @param uri Configures the connection with a mongocxx::uri. It has the username, password, cluster string, 
+     * and the port number to use for connecting.
+     */
+    static void configure(mongocxx::uri uri) 
+    {
+        class noop_logger : public mongocxx::logger 
+        {
+            public:
+                virtual void operator()(mongocxx::log_level,
+                            bsoncxx::stdx::string_view,
+                            bsoncxx::stdx::string_view) noexcept {}
+        };
+
+        auto instance = bsoncxx::stdx::make_unique<mongocxx::instance>(bsoncxx::stdx::make_unique<noop_logger>());
+
+        MongoDBConnection::instance().configure(std::move(instance), bsoncxx::stdx::make_unique<mongocxx::pool>(std::move(uri)));
+    }
+
+    /**
+     * @brief Construct a new MongoDBEmbedFunctionContext object
+     * 
+     * @param _logctx Context logger for use with the MongoDBRecordBinder
+     * MongoDBDatasetBinder classes.
+     * @param options Pointer to the list of options that are passed into the Embed function.
+     * This contains the server name, username, password, etc.
+     * @param _flags Should be zero if the embedded script is ok.
+     */
+    MongoDBEmbedFunctionContext::MongoDBEmbedFunctionContext(const IContextLogger &_logctx, const char *options, unsigned _flags)
+    : logctx(_logctx), m_NextRow(), m_nextParam(0), m_numParams(0), m_scriptFlags(_flags)
+    {
+        const char *server = "";
```

Comment:
MongoDB does accept connection options such as those, but they need to be specified in the connection string which is supplied by the user. Since it has to be up to the user adding this information to the README.md file will be helpful.

Reply:
ok, let's def add that info to README.md, and the examples, but it looks like you're constructing a connection string on line 885 based off input options...

Reply:
Yes that is an oversight on my part. I will add an additional option for the User to specify any number of connection options and add that to the connection string.